### DISCUSSION TITLE
Incomplete bitbucket-server user results in invalid gitAuthor

### DISCUSSION THREAD

**Discussion by Kiskae** [renovatebot/renovate#31174] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

bitbucket-server, renovate 38.52.2

### Please tell us more about your question or problem

Unable to provide a reproduction since it involves just the platform integration.

Root cause appears to be this change: https://github.com/renovatebot/renovate/pull/29525

I've traced it to the fact that the bitbucket server account linked to the renovate token does not have an email address configured, this causes the change above to set the `gitAuthor` to a value without an email and triggers the validation error later on.

Essentially the value `emailAddress` in the following line can be an empty string: 
https://github.com/amezin/renovate/blob/745978f11ffbc2585be20f005fdbe854b19b466c/lib/modules/platform/bitbucket-server/index.ts#L151

### Logs (if relevant)

<details><summary>Logs</summary>

```

[2024-09-03T07:17:48.267Z]          "validationSource": "None",
[2024-09-03T07:17:48.267Z]          "validationError": "Invalid gitAuthor",
[2024-09-03T07:17:48.267Z]          "validationMessage": "`gitAuthor` is not parsed as valid RFC5322 format: `******* <>`",
[2024-09-03T07:17:48.267Z]          "message": "config-validation",

```

</details>

**Comment by rarkins** [comment]
What is the string which is returned?

**Reply by Kiskae** [reply]
From `https://<server>/rest/api/1.0/users/<user>`:
```json
{
    "name": "<user>",
    "emailAddress": "",
    "active": true,
    "displayName": "<user>",
    "id": 1234,
    "slug": "<user>",
    "type": "NORMAL",
    ...
}
```

**Comment by rarkins** [comment]
Converted to issue: https://github.com/renovatebot/renovate/issues/31175
